enabler/src/de/schildbach/pte/NetworkProvider.java

```diff
@@ -68,6 +68,38 @@
         BIKE
     }
 
+    public class TripOptions {
+        public @Nullable Set<Product> products;
```

Comment:
Can we make these fields final?

Reply:
Sure, but then we would have to change some logic where e.g. `options.products` is excepted to be non-null (see [here](https://github.com/schildbach/public-transport-enabler/pull/216/files/13e16d3e4571d96572828b2388dfaeb058aefe6e#diff-b2543011aff601f59664df7f7e17430cR851) and some lines downwards as an example).

Do you want me to amend to this commit or add a new commit with these changes or do you want to change it while merging?

Reply:
Oh yes, indeed. Please do not mutate the TripOptions instance. It's owned by the caller, who will not expect it to be changed as a side effect of calling.